metamorph/processor.go

```diff
@@ -310,35 +278,51 @@ func (p *Processor) statusUpdateWithCallback(statusUpdates []store.UpdateStatus)
 }
 
 func (p *Processor) processExpiredTransactions() {
-	// filterFunc returns true if the transaction has not been seen on the network
-	filterFunc := func(procResp *processor_response.ProcessorResponse) bool {
-		return (procResp.GetStatus() < metamorph_api.Status_SEEN_ON_NETWORK || procResp.GetStatus() == metamorph_api.Status_SEEN_IN_ORPHAN_MEMPOOL) && p.now().Sub(procResp.Start) > unseenTransactionRebroadcastingInterval*time.Second
-	}
+	span, _ := opentracing.StartSpanFromContext(context.Background(), "Processor:processExpiredTransactions")
+	dbctx := opentracing.ContextWithSpan(context.Background(), span)
+	defer span.Finish()
 
-	// Resend transactions that have not been seen on the network
-	// The Items() method will return a copy of the map, so we can iterate over it without locking
+	// Periodically read unmined transactions from database and announce them again
 	for range p.processExpiredTxsTicker.C {
-		expiredTransactionItems := p.ProcessorResponseMap.Items(filterFunc)
+		// define from what point in time we are interested in unmined transactions
+		getUnminedSince := p.now().Add(-1 * p.mapExpiryTime)
+		var offset int64
 
-		if len(expiredTransactionItems) > 0 {
-			p.logger.Info("Resending expired transactions", slog.Int("number", len(expiredTransactionItems)))
-			for txID, item := range expiredTransactionItems {
-				startTime := time.Now()
-				retries := item.GetRetries()
-				item.IncrementRetry()
+		for {
+			// get all transactions since then chunk by chunk
+			fmt.Println(loadUnminedLimit, offset)
+			unminedTxs, err := p.store.GetUnmined(dbctx, getUnminedSince, loadUnminedLimit, offset)
+			if err != nil {
+				p.logger.Error("Failed to get unmined transactions", slog.String("err", err.Error()))
+				continue
+			}
+
+			offset += loadUnminedLimit
+			p.logger.Info("loaded unmined transactions", slog.Int("number", len(unminedTxs)), slog.Int64("limit", loadUnminedLimit))
+			if len(unminedTxs) == 0 {
```

Comment:
The whole for-loop of line 291 will only stop once there are no more unmined txs to be loaded and retried, right?

Reply:
Yeap, exactly